src/libs/actions/Welcome.ts

```diff
@@ -26,21 +37,21 @@ let currentUserAccountID;
  * - Whether we have loaded all reports the server knows about
  */
 function checkOnReady() {
-    if (!_.isBoolean(isFirstTimeNewExpensifyUser) || isLoadingReportData) {
+    if (isFirstTimeNewExpensifyUser === undefined || isLoadingReportData) {
         return;
     }
 
-    resolveIsReadyPromise();
+    resolveIsReadyPromise?.();
 }
 
 Onyx.connect({
     key: ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER,
     initWithStoredValues: false,
-    callback: (val) => {
+    callback: (val: OnyxEntry<boolean>) => {
```

Comment:
```suggestion
    callback: (val) => {
```

You don't need to specify type here, it should be inferred automatically.